src/states/bootstrapping_peer.rs

```diff
@@ -446,11 +443,12 @@ mod tests {
             panic!("Should have received `NewMessage` event.");
         }
 
-        // Drop the network service...
+        // Drop the network service and let some time pass...
         drop(node_a_network_service);
+        FakeClock::advance_time(BOOTSTRAP_TIMEOUT.as_secs() * 1000 + 1);
```

Comment:
why use the bootstrap timeout?

Reply:
Because we want the current bootstrap request to timeout so it tries the next one (which doesn't exist so it terminates).